### DISCUSSION TITLE
Differences of the read and write mode?

### DISCUSSION THREAD

**Discussion by Karol-G** [zarr-developers/zarr-python#1011] [Q&A] [answered]
Hi,

one can set the mode of a zarr array to be read or write (amongst other modes). What are the different effects of such modes?
In particular, if I create a larger-than-RAM array in write-only mode, does it mean that no memory will be occupied by the array?
If yes, is this also true if I write into different chunks in this write-only array?

Best,
Karol

**Comment by joshmoore** [comment, marked as answer]
Hi @Karol-G, I migrated the issue to discussions to match your intent. Hope that's ok.

> What are the different effects of such modes?

This works quite similarly to the modes to `open()`. If a Zarr is opened for a read, then it shouldn't overwrite chunks.

> if I create a larger-than-RAM array in write-only mode, does it mean that no memory will be occupied by the array?

If it's a _Zarr_ array, then yes, the whole array should never be in memory. If you materialize the array, for example by trying to fill a numpy array, then you could (or will!) run out of memory.

> If yes, is this also true if I write into different chunks in this write-only array?

Yes. When you write into the array, each chunk is written to disk as soon as it is full. Again, it's probably more a question of why type (zarr or not zarr) and how large the input array is.

**Reply by Karol-G** [reply]
Thanks a lot. That is what I wanted to know!
